citesphere/src/main/java/edu/asu/diging/citesphere/core/service/giles/impl/GilesUploadCheckerImpl.java

```diff
@@ -140,8 +140,22 @@ public void checkUploads() {
                     // Giles is still procoessing
                     logger.debug("Upload " + upload.getProgressId()
                             + " still being processed.");
-                    checkedUploads.add(upload);
-                    continue;
+                    ObjectMapper mapper = new ObjectMapper();
+                    String jsonBody = response.getBody();
+                    GilesCheckUploadResponse processed = new GilesCheckUploadResponse();
+                    try {
+                        processed = mapper.readValue(jsonBody, GilesCheckUploadResponse.class);
+                    } catch (IOException e) {
+                        logger.error("Could not deserialize response.", e);
+                        upload.setDocumentStatus(GilesStatus.FAILED);
+                        checkedUploads.add(upload);
+                    }
+                    GilesUpload processedUpload = new GilesUpload();
+                    processedUpload.setProgressId(upload.getProgressId());
+                    processedUpload.setUploadId(processed.getUploadId());
```

Comment:
this will throw an error if there was an exception thrown during `processed = mapper.readValue`